tvb_epilepsy/base/h5_model.py

```diff
@@ -19,19 +45,321 @@ def append(self, h5_model):
+                    for class_type, class_instance in children_dict.iteritems():
+                        if class_name == class_type:
+                            child_object = class_instance
+                    if isinstance(child_object, (list, tuple)):
+                        # if it is a list or tuple...
+                        grandchild_name = child_key.split('/', 1)[0]
+                        # but its own children names are not strings of integers:
+                        if not(grandchild_name.isdigit()):
+                            # convert to a dict
+                            child_object = list_or_tuple_to_dict(child_object)
+                        # if it is a tuple...
+                        if isinstance(child_object, tuple):
+                            # ...convert to list that is mutable
+                            child_object = list(child_object)
+                # If still not created, make a dict() by default:
+                if child_object is None:
+                    logger.warning("\n Child object " + str(name) +
+                                   " still not created! Creating an Ordereddict() by default!")
+                    child_object = OrderedDict()
+                # ...and continue to further specify it...
+                children_dict.update(getattr(child_object, "children_dict", {}))
+                build_hierarchical_object_recursively(child_object, child_key, value, children_dict)
+                if class_name == "tuple":
+                    child_object = tuple(child_object)
+                set_field(obj, name, child_object)
+        except:
+            warnings.warn("Failed to set attribute " + str(key) + " of object " + obj.__class__.__name__ + "!")
+
+
+def read_h5_model(path):
+
+    h5_file = h5py.File(path, 'r', libver='latest')
+
+    datasets_dict = dict()
+    metadata_dict = dict()
+
+    for key, value in h5_file.attrs.iteritems():
+        metadata_dict.update({key: value})
+
+    datasets_keys = return_h5_dataset_paths_recursively(h5_file)
+
+    for key in datasets_keys:
+        datasets_dict.update({key: h5_file[key][()]})
+
+    datasets_dict = sort_dict(datasets_dict)
+    metadata_dict = sort_dict(metadata_dict)
+
+    return H5Model(datasets_dict, metadata_dict)
+
+
+def return_h5_dataset_paths_recursively(group):
+    try:
+        paths = []
+        for key in group.keys():
+            paths += return_h5_dataset_paths_recursively(group[key])
+
+        return paths
+    except:
+        return [group.name]
+
+
+if __name__ == "__main__":
+
+    from copy import deepcopy
+
+    from tvb_epilepsy.base.constants import FOLDER_RES, DATA_MODE, DATA_CUSTOM, TVB
+    from tvb_epilepsy.base.utils import assert_equal_objects
+    from tvb_epilepsy.base.model_vep import Connectivity
+    from tvb_epilepsy.base.disease_hypothesis import DiseaseHypothesis
+
+    if DATA_MODE is TVB:
+        from tvb_epilepsy.tvb_api.readers_tvb import TVBReader as Reader
```

Comment:
This dependency is completely wrong. 
We should not import from the other packages in "base"!!